sqlglot/parser.py

```diff
@@ -2482,6 +2577,65 @@ def _parse_query_modifiers(
                 break
         return this
 
+    def _generic_parse_option(
+        self, option_name: str, valid_continuations: t.List[t.Union[str, t.Callable]]
+    ) -> t.Optional[exp.Option]:
+        # Handle the special case of options with no continuations.
+        if not valid_continuations:
+            return self._return_option_with_value(
+                self.expression(exp.Identifier, this=option_name), None
```

Comment:
We shouldn't be constructing `Identifier`s here - in reality these are `Var`s. They represent different things.

Reply:
`Var` represents "keywords", whereas `Identifier` represents names of data objects such as columns, tables, etc. This distinction is important because for example identifiers are transformed by the optimizer whereas vars are not.

Reply:
OK - so we might want to construct a `var` in case of keywords, and a Literal otherwise - WDYT?

Reply:
Var sounds good for keywords, yes. What test case would correspond to a `Literal`?

Reply:
For example, `MAXDOP 15`